src/test/java/reposense/git/GitCheckerTest.java

```diff
@@ -10,24 +9,11 @@
 import org.junit.Assert;
 import org.junit.Test;
 
-import reposense.model.RepoConfiguration;
-import reposense.parser.InvalidLocationException;
 import reposense.template.GitTestTemplate;
-import reposense.util.FileUtil;
 import reposense.util.TestUtil;
 
 
 public class GitCheckerTest extends GitTestTemplate {
-    @Test
-    public void checkout_fromDiskLocation_success() throws GitDownloaderException, IOException,
```

Comment:
This test shouldn't be deleted (this functionality shouldn't have been affected by the current PR change).
Rather, just change the `displayName`/`repoName` of the config to ensure we get the correct path to checkout from the correct location.

Reply:
Actually, this test is redundant now as by default the current working branch from the local repository in disk location is being checked out.

Reply:
>Actually, this test is redundant now as by default the current working branch from the local repository in disk location is being checked out.

This test method tests for the  functionality of `git checkout` with a local disk location rather than a github URL, not whether or not the default cwd is being checked out,

Reply:
Get the following error when I attempt to run this test:
10:19:22 - Cloning ../repos/testrepo-Alpha/testrepo-Alpha...
10:19:22 - Error encountered in Git Cloning, will attempt to continue analyzing

reposense.git.GitDownloaderException: Error returned from command git clone ".../repos/testrepo-Alpha/testrepo-Alpha"on path repos/testrepo-Alpha :
fatal: repository '.../repos/testrepo-Alpha/testrepo-Alpha' does not exist

Not sure how to resolve this error

Reply:
here is the stack trace:
```
10:58:16 - Cloning /Users/apoorvaullas/Desktop/RepoSense/repos/testrepo-Alpha/testrepo-Alpha...
10:58:16 - Error encountered in Git Cloning, will attempt to continue analyzing

reposense.git.GitDownloaderException: Error returned from command git clone "/Users/apoorvaullas/Desktop/RepoSense/repos/testrepo-Alpha/testrepo-Alpha"on path repos/testrepo-Alpha :
fatal: repository '/Users/apoorvaullas/Desktop/RepoSense/repos/testrepo-Alpha/testrepo-Alpha' does not exist
	at reposense.git.GitDownloader.downloadRepo(GitDownloader.java:28)
	at reposense.git.GitCheckerTest.checkout_fromDiskLocation_success(GitCheckerTest.java:26)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)


java.lang.RuntimeException: Error Creating Thread:Cannot run program "bash" (in directory "repos/testrepo-Alpha/testrepo-Alpha"): error=2, No such file or directory

	at reposense.system.CommandRunner.runCommand(CommandRunner.java:148)
	at reposense.system.CommandRunner.checkout(CommandRunner.java:43)
	at reposense.template.GitTestTemplate.after(GitTestTemplate.java:69)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)

```
@eugenepeh what would u suggest i rename DISK_REPO_DISPLAY_NAME to?

Reply:
>My suspicion is the DISK_REPO_DISPLAY_NAME needs to be changed.

Don't think it has to do with the `DISK_REPO_DISPLAY_NAME`, but rather since `getRepoRoot()` has been updated to use the `repoName` instead of the `displayName`, we are now are trying to clone the exact same directory where we are running the `git clone` command, and since `GitDownloader` will delete the `repoRoot` before creating a new 'temp' repo directory, we will delete the directory which we want to clone, thus getting this error.

Actually, I think for this we should have kept the 'repoRoot' a constant value (construct it in the `RepoConfiguration` constructor), instead of it suddenly changing when we change the `displayName` (which is why we required to switch it to use `repoName` instead?) and causing things to go haywire.

This is probably a different issue altogether, so maybe we can just temporary remove this test, and raise an issue to bring it back in the future.

@eugenepeh your opinion on this?

Reply:
> This is probably a different issue altogether, so maybe we can just temporary remove this test, and raise an issue to bring it back in the future.

Yes, this is fine. Please open an issue on that @apoorva17 if you decide to remove the test.